src/st/no1/e_skeleton_ape.c

```diff
@@ -326,4 +366,72 @@ void EntitySkeletonApeBarrel(Entity* self) {
 }
 
 // Skeleton Ape barrel explosion
-INCLUDE_ASM("st/no1/nonmatchings/e_skeleton_ape", func_us_801D544C);
+extern EInit D_us_80180B30;
+extern s32 D_us_8018334C[];
+extern s32 D_us_8018335C[];
+extern u16 D_us_8018336C[];
+extern u16 D_us_80183374[];
+extern s16* D_us_801833BC[];
+
+void func_us_801D544C(Entity* self) {
+    Collider collider;
+    s16 x, y;
+    s16 params;
+    s32 velocityY;
+    s32 posX;
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(D_us_80180B30);
+        self->flags =
+            FLAG_DESTROY_IF_OUT_OF_CAMERA |
+            FLAG_DESTROY_IF_BARELY_OUT_OF_CAMERA | FLAG_POS_CAMERA_LOCKED |
+            FLAG_KEEP_ALIVE_OFFCAMERA | FLAG_UNK_2000;
+        self->hitboxState = 0;
+        params = self->params;
+        posX = D_us_8018334C[params];
+        self->ext.skeletonApeBarrel.unk80 = posX;
+        if (self->facingLeft != 0) {
+            self->ext.skeletonApeBarrel.unk80 = -posX;
+        }
+        self->ext.skeletonApeBarrel.unk80 =
+            (s32)self->ext.skeletonApeBarrel.unk80 + self->velocityX;
+        self->velocityY = D_us_8018335C[params];
+        self->animCurFrame = D_us_8018336C[params];
+        self->ext.skeletonApe.unk7C = D_us_80183374[params];
+        return;
+    case 1:
+        self->velocityX = self->ext.skeletonApeBarrel.unk80;
+        velocityY = self->velocityY;
+        if (UnkCollisionFunc3(D_us_801833BC[self->params]) & 1) {
+            if (velocityY > 0) {
+                velocityY = (s32)(velocityY + ((u32)velocityY >> 0x1F)) >> 1;
+                self->ext.skeletonApeBarrel.unk80 =
+                    ((s32)self->ext.skeletonApeBarrel.unk80 / 2);
+                if (velocityY < 0x4000) {
+                    velocityY = 0;
+                    self->ext.skeletonApeBarrel.unk80 = 0;
+                }
+                self->velocityY = -velocityY;
+                self->velocityX = self->ext.skeletonApeBarrel.unk80;
+            } else {
+                EntityExplosionSpawn(0, 0x655);
```

Comment:
Looks like the second arg to this is a SFX, so this can be mapped using the enum in `sfx.h`.
0x655 should be `SFX_EXPLODE_B`